src/rust/catpowder/win/runtime.rs

```diff
@@ -81,21 +81,23 @@ impl NetworkRuntime for SharedCatpowderRuntime {
     }
 
     /// Transmits a packet.
-    fn transmit<P: PacketBuf>(&mut self, mut pkt: P) {
+    fn transmit<P: PacketBuf>(&mut self, mut pkt: P) -> Result<(), Fail> {
         let header_size: usize = pkt.header_size();
         let body_size: usize = pkt.body_size();
         trace!("transmit(): header_size={:?}, body_size={:?}", header_size, body_size);
 
         if header_size + body_size >= u16::MAX as usize {
-            warn!("packet is too large: {:?}", header_size + body_size);
-            return;
+            let cause = format!("packet is too large: {:?}", header_size + body_size);
+            warn!("{}", cause);
+            return Err(Fail::new(libc::ENOTSUP, &cause));
         }
 
         let mut idx: u32 = 0;
 
         if self.0.borrow_mut().tx.reserve_tx(Self::RING_LENGTH, &mut idx) != Self::RING_LENGTH {
-            warn!("failed to reserve producer space for packet");
-            return;
+            let cause = format!("failed to reserve producer space for packet");
+            warn!("{}", cause);
+            return Err(Fail::new(libc::EINVAL, &cause));
```

Comment:
We failed not becase an invalid argument was supplied, but rather we experienced a runtime error. In my opinion, `EAGAIN` would better reflect that condition.